tools/clang/test/HLSLFileCheck/hlsl/diagnostics/warnings/using.hlsl

```diff
@@ -0,0 +1,27 @@
+// RUN: %dxc -E main -T ps_6_0 %s -HV 2021  | FileCheck -check-prefix=HV2021 %s
+
+// RUN: %dxc -E main -T ps_6_0 %s  | FileCheck %s
+
+// CHECK:keyword 'using' may not be supported by compilers prior to HLSL2021
+// CHECK:error: control reaches end of non-void function
+
+// HV2021-NOT:keyword 'using' may not be supported by compilers prior to HLSL2021
+// HV2021:error: control reaches end of non-void function
+namespace n {
+    using f = float;
+    namespace n2 {
+       f foo(f a) { return sin(a); }
+    }
+
+}
+
+using namespace n;
```

Comment:
Does the warning fire for this line?  It's not clear that it would, and the test does not check that it does.  A verifier test should be better for testing scenarios like this.